libs/core/tests/unit_tests/callbacks/test_async_callback_manager.py

```diff
@@ -0,0 +1,148 @@
+"""Unit tests for verifying event dispatching.
+
+Much of this code is indirectly tested already through many end-to-end tests
+that generate traces based on the callbacks. The traces are all verified
+via snapshot testing (e.g., see unit tests for runnables).
+"""
+
+import contextvars
+from contextlib import asynccontextmanager
+from typing import Any, Optional
+from uuid import UUID
+
+from langchain_core.callbacks import (
+    AsyncCallbackHandler,
+    AsyncCallbackManager,
+    BaseCallbackHandler,
+)
+
+
+async def test_inline_handlers_share_parent_context() -> None:
```

Comment:
@parambharat I added a unit test for a simple test case here and kept the more complex unit test

Reply:
@eyurtsev lgtm